libs/dyn/value.go

```diff
@@ -16,7 +16,12 @@ type Value struct {
 	anchor bool
 }
 
-// NilValue is equal to the zero-value of Value.
+// InvalidValue is equal to the zero-value of Value.
```

Comment:
Based on the purpose of it `EmptyValue` or `ZeroValue` might be a better name, WDYT?

Reply:
That's what it is, essentially, though I'd like to convey the validity of a value.

Both `EmptyValue` and `ZeroValue` sound like they may be valid, but it shouldn't be.

It's also symmetric with the name of the `Kind` and the `IsValid` function:
```go
func (v Value) IsValid() bool {
	return v.k != KindInvalid
}
```

Then when we do some operation (like a get) and verify it returned an actual node in the tree, we can do:
```go
out, _ := v.Get("foo.bar")
if !out.IsValid() {
  // foo.bar doesn't exist
}
```